src/main/java/org/folio/rest/util/ErrorHandler.java

```diff
@@ -0,0 +1,95 @@
+package org.folio.rest.util;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import org.apache.http.HttpStatus;
+import org.folio.rest.exception.InputValidationException;
+import org.folio.rmapi.exception.RMAPIServiceException;
+
+import javax.ws.rs.core.Response;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Utility class for mapping exceptions to response that is passed to io.vertx.core.Handler
+ *
+ * ErrorHandler instance can be configured with error mappers for each Exception type by using add* methods,
+ *
+ * When {@link org.folio.rest.util.ErrorHandler#handle(io.vertx.core.Handler, java.lang.Throwable)} method is called,
+ * first mapper that matches exception type is used to construct javax.ws.rs.core.Response and pass it to io.vertx.core.Handler
+ *
+ * Error mappers are checked in the order they were registered
+ */
+public class ErrorHandler {
+
```

Comment:
Awesome -- thanks for doing this and making the error handling in the impl classes so much cleaner!